phoenix-core/src/main/java/org/apache/phoenix/query/GuidePostsCache.java

```diff
@@ -59,22 +59,36 @@
 
     private final ConnectionQueryServices queryServices;
     private final LoadingCache<GuidePostsKey, GuidePostsInfo> cache;
+    private final ExecutorService executor;
 
     public GuidePostsCache(ConnectionQueryServices queryServices, Configuration config) {
         this.queryServices = Objects.requireNonNull(queryServices);
+
+        // The size of the thread pool used for refreshing cached table stats
+        final int statsCacheThreadPoolSize = config.getInt(
+                QueryServices.STATS_CACHE_THREAD_POOL_SIZE,
+                QueryServicesOptions.DEFAULT_STATS_CACHE_THREAD_POOL_SIZE);
+
+        executor = Executors.newFixedThreadPool(statsCacheThreadPoolSize);
```

Comment:
On a second thought, do we know why do we build the cache even when stats are disabled? Can we skip it or would it result in NPE's all around?

Reply:
OK, I changed to initialize thread pool only if stats are enabled.

To the question "do we know why do we build the cache even when stats are disabled?", my thought is that this is fastest and simplest way to fix it, although it isn't the most efficient way.      Fixing in this way still provides the uniform interface to the callers of the GuidePostCache so the calls don't need to something like "if stats are enabled then call this; otherwise call that". One of efficient and formal ways to fix it is that we define the interface of GuidePostsCache, two classes implement the interface - one is GuidePostsEmptyCache and another is current GuidePostsCache. We create object of GuidePostsEmptyCache when stats are disabled.